src/XliffTasks/Model/XlfDocument.cs

```diff
@@ -155,6 +155,25 @@ public bool Update(TranslatableDocument sourceDocument, string sourceDocumentId)
                     changed = true;
                 }
 
+                // If the source and target require different numbers of formatting items then reset
+                // the target string completely. This avoids problems when the source has been updated
+                // to remove formatting items--when formatting the target string we won't have as many
+                // replacement items as it calls for, leading to an exception.
+                // And if the source string is updated to use _more_ items then formatting with the
+                // target string is likely to produce misleading (or outright meaningless) text. In
+                // either case we lose nothing by just reverting the string until it can be localized
+                // again.
```

Comment:
worth a comment as to why we don't limit ourselves to the case where the source has changed (if above): to catch a handback with wrong placeholder count.

Reply:
nevermind, it is in the block. So this wouldn't catch the bad translation case. Because the source node in xlf would match the resx and we wouldn't get in here.

Reply:
bleagh, wrong again, I think. Yes this method is too complex. Again, my fault.

Reply:
Done.